### ISSUE TITLE
How to add diagnostic ion and how to check the results

### ISSUE THREAD

**Issue by douyue623** [Nesvilab/FragPipe#1651]
**- Upload your log file**
(If a log
[log_2024-06-28_09-55-41.txt](https://github.com/user-attachments/files/16030806/log_2024-06-28_09-55-41.txt)
 file hasn't been generated, go to the 'Run' tab in FragPipe, click 'Export Log', zip the resulting "log_[date_time].txt" file to avoid truncation, then attach the zipped file by drag & drop here.)

**- Describe the issue or question:**
Hi I want to search the diagnostic ion from my samples. My sample is tag-labelled citrullinated peptide. Compared to the arginine, the variable modification is 324.1511. But we noticed there were some diagnostic ions after fragment. The m/z of diagnostic ion is around 197.12. So I set the following search parameters.
mass offset: 0/324.15
diagnostic fragment: 197.12
remaining fragment: 127.0311
I'm not sure if this setting is correct.
And after running, how to check which peptide including the diagnostic ions?

![settings](https://github.com/Nesvilab/FragPipe/assets/174064540/82f6cc14-628a-45a5-b542-1f57baf58782)

**Comment by dpolasky** [comment]
Hi,
Without knowing what your tag is, I can't say for sure that you have the masses correct, but generally it looks like you have a correct setup for this kind of search. Unless you expect there to be lots of multiply-citrullinated peptides, I would maybe recommend removing the variable mod R 324.151 or reducing it to only 1. 

To view which PSMs are found with various diagnostic/remainder ions, use the Diagnostic Feature Extraction section on the PTMs tab and enter the same masses there. It will generate a "diagnosticIons.tsv" file in the ptm-shepherd-output folder with a report of all PSMs and the intensities of all diagnostic/remainder ions you specified for each one. However, you will have to change `Report mass shift as a variable mod` to "No" for that method to work, so it should be done as a separate run from any quant you are trying to do. 

Best,
Dan

**Comment by douyue623** [comment]
Hi Dan,
Thanks for your help! When I set the diagnostic ion, how accurate should it be? How many digits should I have after decimal point? Can I set a ppm range for the diagnostic ion?
Another question would be if I remove the variable modification but use the diagnostic ion for peptide identification, where can I check the modified peptides in the output? I didn't see it... (I attached my current settings here)
![setting1](https://github.com/Nesvilab/FragPipe/assets/174064540/124cae25-e0e5-4917-a24c-b3824424adec)
![setting2](https://github.com/Nesvilab/FragPipe/assets/174064540/3858f571-91ba-4c14-a514-adc2d89fa38f)
![setting3](https://github.com/Nesvilab/FragPipe/assets/174064540/ba21c7eb-4e2c-4649-b717-0da3ca33103d)
![setting4](https://github.com/Nesvilab/FragPipe/assets/174064540/d8e4d3c2-6793-4f12-aee8-6903d9f10074)

Thanks!
Yue

**Comment by dpolasky** [comment]
Hi Yue,

> When I set the diagnostic ion, how accurate should it be? How many digits should I have after decimal point? Can I set a ppm range for the diagnostic ion?
The mass tolerance for diagnostic ion detection is the same as the "Fragment mass tolerance (ppm)" setting in the PTM Profiling section of the PTMs tab. You can set this to whatever you like, but need to make sure that you provide an accurate enough mass to capture an ion within the tolerance (e.g., the ion 100.25 would not be captured with 20ppm tolerance if you provide the mass just as 100). 

> if I remove the variable modification but use the diagnostic ion for peptide identification, where can I check the modified peptides in the output?
If you have set "Report mass diff as a variable mod" in the MSFragger settings, the modified peptides will have modifications reported in the `Assigned Modifications` column. If not, modifications will be reported as mass values in the `Delta Mass` column. 

Best,
Dan

**Comment by douyue623** [comment]
Thanks Dan!
And what does Delta Mass mean?

**Comment by dpolasky** [comment]
Delta mass is the difference between observed and calculated peptide mass, which for a mass offset search is the mass of the mass offset (and the mass error and isotope error, if applicable). 

Best,
Dan

**Comment by douyue623** [comment]
Sounds good. So when we set the mass offset in the advanced options of MSFragger, it is for diagnostic ion or for the modification tag on citrullinated peptides. I am kind of confused. And if it is the mass offset for diagnostic ion, since we have two diagnostic ion, can I set two?

**Comment by dpolasky** [comment]
The mass offset is the modification itself, whereas the diagnostic ion is for determining which spectra to search for that modification. Please see our documentation or the MSFragger labile paper for further details. 

There is no limit to the number of diagnostic ions you can provide. 

Best,
Dan

**Comment by douyue623** [comment]
Hi Dan, 
Thanks so much for your help. Now i have some new problems.
When I want to run it, it shows "Cancelling 19 remaining tasks" and cannot go ahead.
Here is the log files, do you know how to let it work again?
[log_2024-07-03_14-15-59.txt](https://github.com/user-attachments/files/16089168/log_2024-07-03_14-15-59.txt)

**Comment by dpolasky** [comment]
From the log: `It is a closed search while localize_delta_mass = 1. Please change localize_delta_mass to 0.`

It looks like you did not specify any mass offsets in the search settings for MSFragger. Double check the parameters on the MSFragger tab if you want to do a mass offset search (which is needed if you want to use diagnostic ions). Alternatively, if you in fact want to do a closed search, load the default workflow and modify from there. 

Best,
Dan

**Comment by douyue623** [comment]
hi Dan, I just want to try new variable modification. So I chose default. But it still doesn't work. Can you help me check it? 
[log_2024-07-08_11-04-04.txt](https://github.com/user-attachments/files/16131063/log_2024-07-08_11-04-04.txt)


Thanks,
Yue

**Comment by dpolasky** [comment]
From the log:
> restrict_deltamass_to string included capital letters. Please change it to 'all' to prevent interpreting as 'resides A, L, L are allowed'.

If you follow this instruction, it will resolve the issue. 

Best,
Dan

**Comment by douyue623** [comment]
Hi I change "ALL" to "all", but it stop again. I have no idea. Can you help me check again?
[log_2024-07-08_16-14-59.txt](https://github.com/user-attachments/files/16134158/log_2024-07-08_16-14-59.txt)

**Comment by dpolasky** [comment]
This is an issue with MSBooster - for a resolution, please submit it as a new issue since it is not related to the rest of this ticket. 

However, I recommend disabling MSBooster for this search because the spectrum and retention time predictions for your modified peptides are not likely to be good (the modification is not in the training data), and it may result in reduced identification of modified peptides compared to standard scoring. 

Best,
Dan

**Comment by yangkl96** [comment]
Hi Yue,

I agree with Dan that MSBooster may not help much with citrullinated peptides. If you wish to test, please download FragPipe v22, which has un updated MSBooster that resolves this error.

Best,
Kevin

**Comment by douyue623** [comment]
Hi I disabled MSBooster, but it stops again. Could you please help me check it again?

[log_2024-07-09_08-47-18.txt](https://github.com/user-attachments/files/16143456/log_2024-07-09_08-47-18.txt)

Thanks,
Yue

**Comment by fcyu** [comment]
It was due to version incompatibility. Please upgrade FragPipe to 22.0.

Best,

Fengchao

**Comment by douyue623** [comment]
Thanks! I can successfully run it now. 
I tried to use diagnostic ion to search again. But when I set "Restrict delta mass" to R, some peptides without R also show in diagnosticion.tsv file. Obviously they are not citrullinated peptides. And what do the last five column mean in this file. How could I identify citrullinated peptides.
![diagnosticIons](https://github.com/Nesvilab/FragPipe/assets/174064540/e46fb145-30be-4b20-9583-50767b2c5087)
